Add a few text-related generators

voidus opened the pull request:
Based on my fumbling at #52 I added this to my project and thought if might be something you'd want to include.

The code points are taken from Hedgehog.

I'm not sure what an appropriate level of testing for code like this would be. I'd be happy to add some, but I think I need a little guidance on what to test.

edsko commented:
This all looks reasonable enough to me. Some suggestions:

- Rather than making this part of the standard generators exported by `.Generator`, we make this a dedicated module; probably we don't want to pollute the general namespace with all kinds of text related generators (I suspect over time more might be added)
- I have no problem with the dependency on #54 (which I've merged but renamed `chooseAny` to `oneof`), but let's avoid the nix stuff if we can (there are some flake files added in this PR).
- In terms of testing, I would let say, let's add a few tests to `TestSuite/Prop/Generator`, maybe a few shrinking tests or minimal value tests for some of these generators. Having a few checks in there will already prevent a lot of regressions, doesn't have to be anything particularly fancy.

edsko commented:
Thanks for the contribution!

edsko commented:
(I do like the "shrink towards printable" goal you mentioned in #52; that is something we can write a test for.)

voidus commented:
rebased out the accidentally commited `flake.{nix,lock}`, I'll look into the other comments when I find time

voidus commented:
I reworked stuff a little and added tests. (They found issues of course :smiley:)

Still need to have a short think of how to best test `string` and `text`, will look into that later.

edsko commented:
Marked this as WIP. Feel free to ping me when you think this is ready for review :)
